<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  
  <link rel="shortcut icon" href="../../img/favicon.ico">
  <title>naiveNN - Predicting Cancer Drug Response (CDR) using Graph Neural Networks (GNN)</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../../css/theme.css" />
  <link rel="stylesheet" href="../../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css" />
  <link href="../../assets/_mkdocstrings.css" rel="stylesheet" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "naiveNN";
    var mkdocs_page_input_path = "03_model/naiveNN_arch.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../../js/jquery-2.1.1.min.js" defer></script>
  <script src="../../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href="../.." class="icon icon-home"> Predicting Cancer Drug Response (CDR) using Graph Neural Networks (GNN)</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../..">Home</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../../motivation/">Motivation</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../../problem_overview/">Problem Overview</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Synthetic Data</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../../01_synth_data/generation/">Generating a Synthetic Dataset</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../../01_synth_data/data_obj/">The Synthetic gnnCDR Data Object</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../../01_synth_data/data_split/">Creating Train/Test/Val Datasets</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Real Data</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="#">Creating a Global Graph</a>
    <ul>
                <li class="toctree-l2"><a class="reference internal" href="../../02_real_data/fi_graph/">Functional Interaction Network</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../../02_real_data/dti_graph/">Drug-Target Interaction Network</a>
                </li>
    </ul>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="#">Exogenous Features</a>
    <ul>
                <li class="toctree-l2"><a class="reference internal" href="../../02_real_data/overview/">Overview</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../../02_real_data/expression/">Expression</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../../02_real_data/mutation/">Mutation</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../../02_real_data/cnv/">Copy Number Variation (CNV)</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../../02_real_data/methylation/">Methylation</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../../02_real_data/gene_dep/">Gene Dependency</a>
                </li>
    </ul>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Model Architecture</span></p>
                <ul class="current">
                    <li class="toctree-l1"><a class="reference internal" href="../gnnCDR_arch/">gnnCDR</a>
                    </li>
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">naiveNN</a>
    <ul class="current">
    </ul>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../targets_and_metrics/">Choice of Target and Metrics</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../ppi_context_edge_weight_decay/">PPI trainable edge weights</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">User Guide</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../../pages_usage/mkdocs_guides/">Mkdocs Help</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../../pages_usage/syntax_examples/">Mkdocs Syntax</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../..">Predicting Cancer Drug Response (CDR) using Graph Neural Networks (GNN)</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../..">Docs</a> &raquo;</li>
    
      
        
          <li>Model Architecture &raquo;</li>
        
      
    
    <li>naiveNN</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  
  <hr/>
</div>

          <div role="main">
            <div class="section">
              
                <h1 id="naive-neural-network-naivenn">Naive Neural Network (naiveNN)</h1>
<p>This represents the baseline model; which has no way of including prior knowledge of protein-protein interactions (PPIs). </p>
<p>For the most part, this can be conceptualized as a simple feed forward neural network; however, we do employ a similar tactic to <code>gnnCDR</code> of fist predicting a <em>time-agnostic</em> latent representation for each gene and then using a shared set of parameters to predict a time series. </p>
<p align="center">
    <img src="../naiveNN_model.PNG" alt="..." width=300>
</p>

<p>The <code>expr (t=0)</code> input represents the gene expression abundance at baseline (no drug), and is the expression value at time=0 in each expression time series. The <code>drug</code> input identifies the presence of a given perturbation - chemical perturbations are active when they have non-zero values, and which represent drug concentration. Genetic perturbations are specified as active with an arbitrary value of 1. A feed forward fully connected neural network then predicts a latent representation <code>z</code> for every gene. This is intended to represent a time-agonistic representation of that genes expression time series, e.g., response to the perturbagen. The time-agnostic latent vector of each gene is then treated as input along with a specific time point (<code>time</code> in the figure) to a second neural network, which predicts a discrete expression value for every gene. </p>
<p>In this way, our model shares the time-series function across all genes. This was found to be a significant improvement over a classical neural network (e.g., no time-agnostic latent prediction or shared time-series function). </p>
<p>For convenience, we've added the data structures: </p>
<pre><code>data.baseline 
data.x 
data.y 
data.time
</code></pre>
<p>To the <code>SynthHeteroDataset</code> HeteroData object, which provides the relevant data in (observation, feature) views. </p>
<h1 id="example">Example</h1>
<p>To initialize a model: </p>
<pre><code class="language-python">
data = train_dataset.__getitem__(0)                                     # HeteroData object from SynthHeteroDataset

in_channels         = data['protein'].num_nodes + data.pert_all.size(1) # input channels to neural network (not including time)
out_channels        = data['protein'].num_nodes                         # number of genes 
dropout             = 0.15                                              # regularization parameter 
num_layers          = 2                                                 # number of fully connected layers
hidden_channels     = 100                                               # number of hidden units in each layer 

model = gnn_cdr.model.NaiveNN(in_channels=in_channels, 
                              out_channels=out_channels, 
                              hidden_channels=hidden_channels, 
                              num_layers=num_layers, 
                              dropout=dropout)

</code></pre>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../targets_and_metrics/" class="btn btn-neutral float-right" title="Choice of Target and Metrics">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../gnnCDR_arch/" class="btn btn-neutral" title="gnnCDR"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
      <p>Copyright &copy; 2021 Nate Evans</p>
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
    
      <span><a href="../gnnCDR_arch/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../targets_and_metrics/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script>var base_url = '../..';</script>
    <script src="../../js/theme_extra.js" defer></script>
    <script src="../../js/theme.js" defer></script>
      <script src="https://polyfill.io/v3/polyfill.min.js?features=es6" defer></script>
      <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js" defer></script>
      <script src="../../javascripts/mathjax.js" defer></script>
      <script src="../../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
