<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <link rel="shortcut icon" href="../../img/favicon.ico" />
    <title>Expression - Predicting Cancer Drug Response (CDR) using Graph Neural Networks (GNN)</title>
    <link rel="stylesheet" href="../../css/theme.css" />
    <link rel="stylesheet" href="../../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Expression";
        var mkdocs_page_input_path = "02_real_data/expression.md";
        var mkdocs_page_url = null;
      </script>
    
    <script src="../../js/jquery-3.6.0.min.js" defer></script>
    <!--[if lt IE 9]>
      <script src="../../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
      <script>hljs.initHighlightingOnLoad();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="../.." class="icon icon-home"> Predicting Cancer Drug Response (CDR) using Graph Neural Networks (GNN)
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../..">Home</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../../motivation/">Motivation</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../../problem_overview/">Problem Overview</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">Synthetic Data</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../01_synth_data/generation/">Generating a Synthetic Dataset</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../01_synth_data/data_obj/">The Synthetic gnnCDR Data Object</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../01_synth_data/data_split/">Creating Train/Test/Val Datasets</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Real Data</span></p>
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="#">Creating a Global Graph</a>
    <ul>
                <li class="toctree-l2"><a class="reference internal" href="../fi_graph/">Functional Interaction Network</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../dti_graph/">Drug-Target Interaction Network</a>
                </li>
    </ul>
                  </li>
                  <li class="toctree-l1 current"><a class="reference internal current" href="#">Exogenous Features</a>
    <ul class="current">
                <li class="toctree-l2"><a class="reference internal" href="../overview/">Overview</a>
                </li>
                <li class="toctree-l2 current"><a class="reference internal current" href="./">Expression</a>
    <ul class="current">
    <li class="toctree-l3"><a class="reference internal" href="#data-source-and-info">Data source and info</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#reactome-fi-coverage">Reactome FI Coverage</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#lincs-coverage">LINCS Coverage</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#z-score-transformation">Z-score transformation</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#accessing-individual-cell-line-expression">Accessing individual cell line expression</a>
    </li>
    </ul>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../mutation/">Mutation</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../cnv/">Copy Number Variation (CNV)</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../methylation/">Methylation</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../gene_dep/">Gene Dependency</a>
                </li>
    </ul>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Model Architecture</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../03_model/gnnCDR_arch/">gnnCDR</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../03_model/naiveNN_arch/">naiveNN</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../03_model/targets_and_metrics/">Choice of Target and Metrics</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../03_model/ppi_context_edge_weight_decay/">PPI trainable edge weights</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Parking Lot</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../04_parking_lot/prism/">Predicting Cell Viability</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">User Guide</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../pages_usage/mkdocs_guides/">Mkdocs Help</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../pages_usage/syntax_examples/">Mkdocs Syntax</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../..">Predicting Cancer Drug Response (CDR) using Graph Neural Networks (GNN)</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../.." class="icon icon-home" alt="Docs"></a> &raquo;</li>
          <li>Real Data &raquo;</li>
          <li>Exogenous Features &raquo;</li><li>Expression</li>
    <li class="wy-breadcrumbs-aside">
    </li>
  </ul>
  <hr/>
</div>

          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="cancer-cell-line-encyclopedia-unperturbed-rna-seq-expression">Cancer Cell Line Encyclopedia unperturbed RNA-seq expression</h1>
<h2 id="data-source-and-info">Data source and info</h2>
<p>We will use the CCLE expression dataset, which can be downloaded <a href="https://ndownloader.figshare.com/files/34008404">here</a>, more info <a href="https://depmap.org/portal/download/">here</a>. </p>
<pre><code>From DepMap: 

Gene expression TPM values of the protein coding genes for DepMap cell lines. Values are inferred from RNA-seq data using the RSEM tool and are reported after log2 transformation, using a pseudo-count of 1; log2(TPM+1).

Additional RNA-seq-based expression measurements are available for download as part of the full DepMap Data Release

More information on the DepMap Omics processing pipeline is available at https://github.com/broadinstitute/depmap_omics.


Genes:19177
Cell Lines:1393
Primary Diseases:33
Lineages:38
Source: Broad Institute
</code></pre>
<h2 id="reactome-fi-coverage">Reactome FI Coverage</h2>
<p>There are <code>403</code> ccle gene's lost in the mapping from gene symbol to ensembl. </p>
<p>13373 of 13609 reactome FI genes have coverage in CCLE expression dataset (98% coverage). </p>
<h2 id="lincs-coverage">LINCS Coverage</h2>
<p>Of the total cell lines in LINCS (n=239), 133 of them have CCLE expression data (56%); However, these 133 lines make up the majority of observations in LINCS L1000. After filtering observations that did not pass QC, these 133 lines with CCLE expression coverage made up 1.83/2.16 million LINCS observations (85%). </p>
<h2 id="z-score-transformation">Z-score transformation</h2>
<p>To make imputation and gene-to-gene expression more comparable for our GNN we will perform z-score tranformations: </p>
<div class="arithmatex">\[ zscore(x) = \frac{x - \bar{x}}{\sigma^2} \]</div>
<p>Here is a visualization of the original and transformed distributions: </p>
<p><img alt="" src="../expr_gene_trans_dist.png" /></p>
<p>This can be done using the function: </p>
<pre><code class="language-python">expr = gnn_cdr.depmap.load_ccle_expr(zscore=True)
</code></pre>
<h2 id="accessing-individual-cell-line-expression">Accessing individual cell line expression</h2>
<p>We can get a specific cell lines expression by using the method: </p>
<pre><code class="language-python">x = gnn_cdr.depmap.get_cell_line_ccle_expr(expr, cell_line, genelist)
</code></pre>
<p>where: </p>
<ul>
<li><code>expr</code> is the pandas dataframe returned by <code>load_ccle_expr</code></li>
<li><code>cell_line</code> is the cell_iname identifier </li>
<li><code>genelist</code> is the list of genes to return; note, genes that are in genelist but not in expr will be <em>zero</em> imputed. </li>
</ul>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../overview/" class="btn btn-neutral float-left" title="Overview"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../mutation/" class="btn btn-neutral float-right" title="Mutation">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
      <p>Copyright &copy; 2021 Nate Evans</p>
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
    
      <span><a href="../overview/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../mutation/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script>var base_url = '../..';</script>
    <script src="../../js/theme_extra.js" defer></script>
    <script src="../../js/theme.js" defer></script>
      <script src="https://polyfill.io/v3/polyfill.min.js?features=es6" defer></script>
      <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js" defer></script>
      <script src="../../javascripts/mathjax.js" defer></script>
      <script src="../../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
